.github/workflows/release.yml

```diff
@@ -0,0 +1,45 @@
+name: Release
+
+on:
+  release:
+    types:
+      - created
+      - edited
```

Comment:
I would not run it in case the release is edited, sometimes we change some lines in the changelog and I would avoid re-running the pipeline :)

Reply:
(but maybe you are just using it to test without creating many releases. In that case my bad 😅)

Reply:
I wonder if removing the `edited` event would cause the transition from pre-release to release to be missed. Have you checked this?

Additionally another problem of missing the edit event would be if you change the tag of the release (perhaps because of a typo). At least I would expect that changing a release would re-run everything just in case.

Reply:
Following internal discussion, we have decided that:

- Editing a release will not re-run the release pipelines, as some services (e.g. Docker) may refuse to overwrite a tag causing the pipeline to unexpectedly fail
- Pre-releases will still push artifacts, but:
  + Docker image tags will have a `-pre` suffix
  + Artifacts will be uploaded to a different s3 repo if it exists, and will have a `-pre` suffix if it does not